scirpy/io/_convert_anndata.py

```diff
@@ -174,28 +133,37 @@ def to_ir_objs(adata: AnnData) -> List[IrCell]:
 
     Returns
     -------
-    List of IrCells
+    `Dandelion` object.
+
     """
-    cells = []
     try:
-        for cell_id, row in adata.obs.iterrows():
-            tmp_ir_cell = IrCell(cell_id, multi_chain=row["multi_chain"])
-            for chain_type, chain_id in itertools.product(["VJ", "VDJ"], ["1", "2"]):
-                chain_dict = {
-                    key: row[f"IR_{chain_type}_{chain_id}_{key}"] for key in IR_OBS_KEYS
+        import dandelion as ddl
+    except:
+        raise ImportError("Please install dandelion: pip install sc-dandelion.")
+    ircelllist = to_ir_objs(adata)
+
+    contig_dicts = {}
+    for ix in ircelllist:
+        for counter, c in enumerate(ix.chains, start=1):
+            contig_dict = {}
+            cell_id = ix.cell_id
+            sequence_id = cell_id + "_contig_" + str(counter)
+            contig_dict.update(
+                {
+                    "cell_id": cell_id,
+                    "sequence_id": sequence_id,
+                    "locus": c.locus,
+                    "junction_aa": c.cdr3,
+                    "junction": c.cdr3_nt,
+                    "umi_count": c.expr,
```

Comment:
@zktuong, should `umi_count` be `duplicate_count`?

Reply:
To be 100% compliant with AIRR, yes. Dandelion allows both and will check for presence of `umi_count` first before checking for `duplicate_count` but i can/should update this to just stick to duplicate_count.

Reply:
Ok, so it doesn't hurt to use `duplicate_count` here in any case. 

How common is the `umi_count` column anyway? I was considering not to support it as it makes stuff more complicated and people could always manually rename it to `duplicate_count` if they have a non-compliant file.

Reply:
I think that `umi_count` is more intuitive than `duplicate_count` but that personal preference. I think the immcantation tools parse it as `umi_count` whereas the 10x AIRR table (and rightly so) is provided as `duplicate_count`. Most people would still be using immcantation-processed files so there would certainly be interest in easy porting between formats but like you said, they can just change their column name first before trying and this will free up time and effort on your end.